### DISCUSSION TITLE
Extend Histogram to work all image sequences

### DISCUSSION THREAD

**Discussion by BrundleThwaite** [opentoonz/opentoonz#6468] [Ideas]
The histogram button in the view only works in the popup view that you get with a rendered sequence and not with a preview render in the viewport.

**Comment by RodneyBaker** [comment]
@BrundleThwaite I hadn't realized that someone had posted a report about the Histogram.

Of late I was under the impression the Histogram wasn't working at all and wondered if perhaps the code hadn't been brought back up to date from Studio Ghibli edition.
Further testing revealed what you observe... that the Histogram only appears to work with previously rendered image/sequences.

A question:  What is the intended purpose of the Histogram in OpenToonz and is it serving that purpose?

I see two entries in the OpenToonz documentation for the Histogram.  One for rendered sequences and the other for scanned images/sequences in Cleanup:

https://opentoonz.readthedocs.io/en/latest/search.html?q=histogram&check_keywords=yes&area=default

**Comment by gab3d** [comment]
Histogram also works when using the "View..." option in the Xsheet (RMB menu), to view a level using the Flipbook.

With respect to the first Histogram appeareance in the manual (related to the Clean-up process), this was related to a Toonz option for Line Processing greyscale images, called "Autoadjust", that didn't make it into OpenToonz. So it should be removed from the manual, I guess (I could do that).

**Comment by RodneyBaker** [comment]
Ah yes.  Thanks @gab3d  Flipbook!  That was the part that I had forgotten.  A reference to use of the Flipbook was what clued me in that the Histogram was actually working.

Question (before you remove any references from the documentation):  Might Autoadjust be turned back on in OpenToonz or was that superseded by GTS and other Ghibli Studio based cleanup?

**Comment by RodneyBaker** [comment]
Follow up to this issue.
Perhaps a check could be added to the triggering of the Histogram that determines if the image has been rendered and A) Turn the icon grey/inactive if false B) Add a message to instruct the user to Preview/Render C) Both   D) Something else.

Alternatively, and perhaps better (?) perhaps triggering the Histogram would automatically launch the current frame in a Flipbook window which would then allow the Histogram to be populated.  I don't prefer this because there are certainly cases where we want to view the Histogram while in the Viewer or ComboViewer.  The user just needs to know that they need to render/preview first.

Perhaps a textual note could be added to the Histogram dialog.

**Comment by RodneyBaker** [comment]
In light of recent updates to Histogram we need to confirm the Histogram still does not work with rendered image sequences as outlined in this report.
